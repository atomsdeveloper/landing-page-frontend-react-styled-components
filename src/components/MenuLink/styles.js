import styled, { css } from 'styled-components';

const Container = styled.a`
    ${({theme}) => css`
        display: block;
        text-decoration: none;
        font-size: ${theme.fontSize.default};
        padding: ${theme.fontSize.default};
        color: ${theme.colors.textColorDark};
        position: relative;

        &::after {
            content: '';
            position: absolute;
            bottom: : 0;
            left: 0;
            width: 0;
            height: 0.2rem;
            background: ${theme.colors.textColorLight};
        }

        &:hover::after {
            left: 25%;
            width: 50%;
        }
    `}

`;
export default Container;