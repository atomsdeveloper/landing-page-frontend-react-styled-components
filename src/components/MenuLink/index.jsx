import P from 'proto-types';
import * as Styled from './styles';

const MenuLink = ( {children, link, newTab = false} ) => {
    const target = newTab ? '_blank' : '_self';
    return (
        <Styled.MenuLink href={link} newTab={target}>
            { children }
        </Styled.MenuLink>
    );
};
MenuLink.propTypes = {
    children: P.string.isRequired,
    link: P.string.isRequired,
    newTab: P.bool.isRequired
};
export default MenuLink;