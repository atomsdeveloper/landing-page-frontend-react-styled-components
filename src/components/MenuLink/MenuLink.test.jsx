import { screen } from '@testing-library/dom';
import { render } from '@testing-library/react';

import MenuLink  from '.';

describe('<MenuLink />', () => {
    it('should render a link', () => {
        render(<MenuLink link='https://www.google.com.br/'> Children </MenuLink>);
        expect(screen.getByRole('link', {name: 'children'}).toBeInTheDocument());
    });

    it('should render a link with target self or blank', () => {
        render(<MenuLink target='_self'> Children </MenuLink>);
        expect(screen.getByRole('target', {name: '_self' || '_black'}).toBeInTheDocument());
    });

    it('should render a link with name value', () => {
        render(<MenuLink target='_self'> Children </MenuLink>);
        expect(screen.getByRole('target', {name: '_self' || '_black'}).firstChild.toBeInTheDocument());
    })
});