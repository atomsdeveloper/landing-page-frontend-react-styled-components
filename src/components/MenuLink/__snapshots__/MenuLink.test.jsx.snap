// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should render open in a new tab 1`] = `
.c0 {
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0F2B46;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: :0.76rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #ffff;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

<a
  class="c0"
  href="http://localhost"
  target="_blank"
>
  Children
</a>
`;
